sdks/go/pkg/beam/core/runtime/exec/sdf.go

```diff
@@ -664,6 +686,21 @@ func (n *ProcessSizedElementsAndRestrictions) GetInputId() string {
 	return indexToInputId(0)
 }
 
+// GetOutputWatermark gets the current output watermark of the splittable unit
+// if one is defined, or returns nil otherwise.
+func (n *ProcessSizedElementsAndRestrictions) GetOutputWatermark() map[string]*timestamppb.Timestamp {
+	if n.PDo.we != nil {
+		ow := timestamppb.New(n.PDo.we.CurrentWatermark())
+		owMap := make(map[string]*timestamppb.Timestamp)
+		for _, out := range n.outputs {
+			owMap[out] = ow
```

Comment:
Maybe it's because I'm not well-versed in how watermarks work, but why store output watermarks in a map like this if we set the same value for every single output? Why not just store the single timestamp? Is it because of some upcoming work that's going to build on this?

Reply:
I can't speak to why this decision was originally made (though I imagine it makes things significantly easier for the runner since input watermark is an important concept and := `min(all incoming output watermarks)`). It is worth noting that we're not actually setting _every_ output to this value, we're just setting the outputs of this transform to this value. At the runner level, I believe this is unioned with other output watermarks that may exist and this map is used to make triggering/windowing decisions